tests/integration/operatorhub/cmd_link_test.go

```diff
@@ -0,0 +1,93 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo link command tests for OperatorHub", func() {
+
+	var commonVar helper.CommonVar
+	var oc helper.OcRunner
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		helper.Chdir(commonVar.Context)
+		oc = helper.NewOcRunner("oc")
+	})
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	Context("Operators are installed in the cluster", func() {
+
+		var etcdOperator string
+		var etcdCluster string
+
+		BeforeEach(func() {
+			// wait till odo can see that all operators installed by setup script in the namespace
+			odoArgs := []string{"catalog", "list", "services"}
+			operators := []string{"etcdoperator", "service-binding-operator"}
+			for _, operator := range operators {
+				helper.WaitForCmdOut("odo", odoArgs, 5, true, func(output string) bool {
+					return strings.Contains(output, operator)
+				})
+			}
+
+			list := helper.CmdShouldPass("odo", "catalog", "list", "services")
+			etcdOperator = regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]-clusterwide`).FindString(list)
+			etcdCluster = fmt.Sprintf("%s/EtcdCluster", etcdOperator)
+		})
+
+		AfterEach(func() {
+			helper.DeleteProject(commonVar.Project)
+		})
+
+		When("a component and a service are deployed", func() {
+
+			var componentName string
+			var svcFullName string
+
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+				componentName = "cmp-" + helper.RandString(6)
+				helper.CmdShouldPass("odo", "create", "nodejs", componentName)
+
+				serviceName := "service-" + helper.RandString(6)
+				svcFullName = strings.Join([]string{"EtcdCluster", serviceName}, "/")
+				helper.CmdShouldPass("odo", "service", "create", etcdCluster, serviceName, "--project", commonVar.Project)
+
+				helper.CmdShouldPass("odo", "push")
+				oc.PodsShouldBeRunning(commonVar.Project, componentName+`-app-.[a-z0-9-]*`)
+			})
+
+			It("should find files in component container", func() {
+				helper.CmdShouldPass("odo", "exec", "--", "ls", "/project/server.js")
+			})
+
+			When("a link is created between the component and the service", func() {
+
+				BeforeEach(func() {
+					helper.CmdShouldPass("odo", "link", svcFullName)
+				})
+
+				When("odo push is executed", func() {
+					BeforeEach(func() {
+						helper.CmdShouldPass("odo", "push")
+						oc.PodsShouldBeRunning(commonVar.Project, componentName+`-app-.[a-z0-9-]*`)
+					})
```

Comment:
@feloy why are we separating `odo link`  and `odo push` in two `when` do we want to add a test in between these two scenario? 
```
When("a link is created between the component and the service and odo push is executed", func() {
				BeforeEach(func() {
					helper.CmdShouldPass("odo", "link", svcFullName)
					helper.CmdShouldPass("odo", "push")
					oc.PodsShouldBeRunning(commonVar.Project, componentName+`-app-.[a-z0-9-]*`)
				})
```

Reply:
Yes, no problem, we can use your version, as there is no test between the two steps.

Reply:
Thanks.